ModelImporter.cpp

```diff
@@ -85,12 +85,23 @@ Status importInput(ImporterContext* importer_ctx,
 
 Status importInputs(ImporterContext* importer_ctx,
                     ::ONNX_NAMESPACE::GraphProto const& graph,
-                    string_map<TensorOrWeights>* tensors) {
+                    string_map<TensorOrWeights>* tensors,
+                    uint32_t weights_count,
+                    onnxTensorDescriptor const* weight_descriptors) {
+  // The weights may come from two sources:
+  // 1. Initializer list in onnx graph
+  // 2. User specified weight through onnxifi
   string_map<::ONNX_NAMESPACE::TensorProto const*> initializer_map;
   for( ::ONNX_NAMESPACE::TensorProto const& initializer : graph.initializer() ) {
     ASSERT(!initializer_map.count(initializer.name()), ErrorCode::kINVALID_GRAPH);
     initializer_map.insert({initializer.name(), &initializer});
   }
+  string_map<onnxTensorDescriptor const*> weight_map;
```

Comment:
You don't validate that tensor names (either in initializers or in weight descriptors) are unique. This may lead to hard-to-spot bugs.

Reply:
OK. I'll check that.